components/espressif/esp32-wroom-32.stanza

```diff
@@ -0,0 +1,277 @@
+; When renaming the stanza package name below, also rename it in the folder stanza.proj
+; See docs.jitx.com for help
+#use-added-syntax(jitx)
+defpackage ocdb/espressif/esp32-wroom-32:
+  import core
+  import collections
+  import jitx
+
+  import jitx/commands
+  import ocdb/defaults
+  import ocdb/land-patterns
+  import ocdb/tolerance
+  import ocdb/box-symbol
+  import ocdb/bundles
+  import ocdb/land-protrusions
+  import ocdb/design-vars
+  import ocdb/generic-components
+  import ocdb/property-structs
+
+; ================================================== Component ==================================================
+public defn component () :
+  component(4,true,false) 
+
+public defn component (pcb-antenna?:True|False) :
+  component(4,pcb-antenna?, false) 
+
+public defn component (memory:Int, pcb-antenna?:True|False) :
+  component(memory, pcb-antenna?, false) 
+
+public pcb-component component (memory:Int, pcb-antenna?:True|False, high-temp:True|False) :
+  var mem-text:String
+  switch {memory == _} :
+    4 : mem-text = ", 4MB"
+    8 : mem-text = ", 8MB"
+    16 : mem-text = ", 16MB"
+    else: fatal("Invalid memory size of %_ passed to esp32-wroom-32. Valid sizes are 4, 8, and 16" % [memory])
+
+  var package-string = "UE"
+  var package-desc = ", UMCX"
+  if pcb-antenna? :
+    package-string = "E" 
+    package-desc = ", PCB Antenna"
+
+  var temp-text = "N"
+  if high-temp:
+    if memory == 4 :
```

Comment:
Normally for cases like this we'd prefer to have an `ensure-...` function that is called first at the top of the function to validate the parameters of the generator, since it can be difficult to validate the conditions in one place, which is really easy to check (and extend).

Reply:
that has more lines of code in this case.